server/plugins/BelaUGens.cpp

```diff
@@ -0,0 +1,1157 @@
+/*
+    SuperCollider real time audio synthesis system
+    Copyright (c) 2002 James McCartney. All rights reserved.
+    http://www.audiosynth.com
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+*/
+
+/*
+ *  BELA I/O UGens created by nescivi, (c) 2016
+ *  https://www.nescivi.eu
+ */
+
+#include <atomic>
+
+#include "Bela.h"
+// These functions are provided by xenomai
+int rt_printf(const char* format, ...);
+int rt_fprintf(FILE* stream, const char* format, ...);
+
+#include "SC_PlugIn.h"
+#define BELA_SIMPLIFIED_DIGITALIO
```

Comment:
when would someone undef this? i would suggest we either make it a build option in cmake or remove the code hidden by this entirely.

Reply:
It was a pretty big overhaul I did  at some point and I only did as much testing as my very limited knowledge of Sc allowed me to. So I left the old code in there so that if in the future problems cropped up I'd first check against the old version. It can probably be removed now I guess. It's probably easier to spot and fix a mistake in the new code than in the old one anyhow :)